### ISSUE TITLE
Cross-database transaction BUG

### ISSUE THREAD

**Issue by fish-li** [mysql-net/MySqlConnector#794]
MySqlConnector has a bug, if start a transcation, i can't change database.

the same code, MySql.Data is OK

``` cs
[TestMethod]
public void Test_Cross_database_transaction()
{
    Assembly asm = typeof(MySql.Data.MySqlClient.MySqlConnection).Assembly;

    if( asm.Location.EndsWith("MySqlConnector.dll") ) {
        Console.WriteLine("MySqlConnector has a bug, can't support this test.");
        return;
    }

    // MySql.Data support this test

    using( DbContext dbContext = DbContext.Create("mysql") ) {

        dbContext.BeginTransaction();

        var args1 = new { p1 = "abc" };
        dbContext.CPQuery.Create("insert into Trans1 (StrValue) values ( @p1 )", args1).ExecuteNonQuery();


        dbContext.ChangeDatabase("TestTrans");  // switch to another database

        var args2 = new { p1 = 123 };
        dbContext.CPQuery.Create("insert into Trans2 (IntValue) values ( @p1 )", args2).ExecuteNonQuery();

        // if MySqlConnector  ExecuteNonQuery(), throw a error:
        // System.InvalidOperationException: The transaction associated with this command is not the connection's active transaction; see https://fl.vu/mysql-trans

        dbContext.Commit();
    }
}
```

**Comment by bgrainger** [comment]
The current implementation of `ChangeDatabase` "closes" the current database, including the active transaction: https://github.com/mysql-net/MySqlConnector/blob/578c993da6d37e8bfedb2611e057ba43d4446ba1/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs#L249

I'm not entirely sure why this is; it may be a bug.

**Comment by bgrainger** [comment]
Fixed in [0.63.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.63.1). Thanks for reporting it!
